src/main/java/de/triology/cas/ldap/ConnectionAwareAuthenticator.java

```diff
@@ -45,6 +47,17 @@ public ConnectionAwareAuthenticator(
         this.useUserConnectionToFetchAttributes = useUserConnectionToFetchAttributes;
     }
     
+    public void someBuggyMethod() {
+        LdapEntry entry = null;       
+        if (entry == null) {
+            System.out.println(entry.getAttribute());
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER') NullPointerException might be thrown as 'entry' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://ecosystem.cloudogu.com/sonar/coding_rules#rule_key=squid%3AS2259)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://ecosystem.cloudogu.com/sonar/coding_rules#rule_key=squid%3AS106)